test/parallel/test-net-connect-options-port.js

```diff
@@ -83,7 +83,7 @@ const net = require('net');
     }
   }, expectedConnections));
 
-  server.listen(0, 'localhost', common.mustCall(() => {
+  server.listen(0, '127.0.0.1', common.mustCall(() => {
```

Comment:
That's interesting, I missed that information. It makes more sense now. It's a bit concerning that this change may break code using `localhost` to refer to the loopback address, we should probably document somewhere how to fix this issue (by updating the `/etc/hosts` file I guess?).

Reply:
That's a good question and I am not sure how this will play out in production outside of the test suite. If everyone would have the `hosts`-file ready for IPv6 (i.e. including `::1`) that should be okay. I seem to understand this is why this PR should land in a major version. Maybe even the DNS lib needs updating? I don't understand why it uses different methods.

See https://github.com/nodejs/node/blob/master/lib/net.js and look for `dns.lookup`. The code around it for `lookupAndListen` and `lookupAndConnect` is different, though I haven't really understood what the differences do.

On the other hand, not setting `verbatim=true` by default seriously hurts IPv6 adoption. Currently, as long as there's an A record available, NodeJS will always default to the IPv4 address. Unless the dev using the lib manually sets it to true (which almost nobody does). Of course the best would be if listening on `localhost` would open a dualstack socket on both `127.0.0.1` and `::1`. At the moment, this only works for `0.0.0.0`.